### ISSUE TITLE
Crash on worldgen in 1.13-pre5/pre6

### ISSUE THREAD

**Issue by dmr-loggly** [toolbox4minecraft/amidst#403]
Branch: 1-13-snapshots
Operating system: OSX

Steps:
1) Launch
2) Select 1.13-pre5/pre6 profile
3) Press ⌘+R to generate a random world
4) Dismiss error popup
5) Press ⌘+R again

Results:
* First generation results in a recoverable error popup
* Second generation results in hard crash
* Errors are different on first and second tries

Here's a log. The errors are different in each case: 

```
[info] Amidst v4.3-beta3
[info] Current system time: 2018-07-04 12:29:51.767
[info] System.getProperty("os.name") == 'Mac OS X'
[info] System.getProperty("os.version") == '10.13.4'
[info] System.getProperty("os.arch") == 'x86_64'
[info] System.getProperty("java.version") == '10.0.1'
[info] System.getProperty("java.vendor") == 'Oracle Corporation'
[info] System.getProperty("sun.arch.data.model") == '64'
[info] Enabling OpenGL.
[info] Biome profile activated: default
[info] using '.minecraft' directory at: '/Users/dmaxwell-ross/Library/Application Support/minecraft'
[info] using biome profiles at: 'biome'
[info] Starting to download remote version list.
[info] Scanning for profiles.
[info] Successfully loaded remote version list.
[info] There are no updates available.
[info] Successfully loaded profile list.
[info] using minecraft launcher profile 'Latest snapshot' with versionId '1.13-pre5'
[info] Loading libraries.
[info] Found library com.mojang:patchy:1.1 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/mojang/patchy/1.1/patchy-1.1.jar
[info] Found library oshi-project:oshi-core:1.1 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/oshi-project/oshi-core/1.1/oshi-core-1.1.jar
[info] Found library net.java.dev.jna:jna:4.4.0 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/net/java/dev/jna/jna/4.4.0/jna-4.4.0.jar
[info] Found library net.java.dev.jna:platform:3.4.0 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/net/java/dev/jna/platform/3.4.0/platform-3.4.0.jar
[info] Found library com.ibm.icu:icu4j-core-mojang:51.2 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/ibm/icu/icu4j-core-mojang/51.2/icu4j-core-mojang-51.2.jar
[info] Found library net.sf.jopt-simple:jopt-simple:5.0.3 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/net/sf/jopt-simple/jopt-simple/5.0.3/jopt-simple-5.0.3.jar
[info] Found library com.paulscode:codecjorbis:20101023 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/paulscode/codecjorbis/20101023/codecjorbis-20101023.jar
[info] Found library com.paulscode:codecwav:20101023 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/paulscode/codecwav/20101023/codecwav-20101023.jar
[info] Found library com.paulscode:libraryjavasound:20101123 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/paulscode/libraryjavasound/20101123/libraryjavasound-20101123.jar
[info] Found library com.paulscode:soundsystem:20120107 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/paulscode/soundsystem/20120107/soundsystem-20120107.jar
[info] Found library io.netty:netty-all:4.1.25.Final at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/io/netty/netty-all/4.1.25.Final/netty-all-4.1.25.Final.jar
[info] Found library com.google.guava:guava:21.0 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/google/guava/guava/21.0/guava-21.0.jar
[info] Found library org.apache.commons:commons-lang3:3.5 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5.jar
[info] Found library commons-io:commons-io:2.5 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/commons-io/commons-io/2.5/commons-io-2.5.jar
[info] Found library commons-codec:commons-codec:1.10 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/commons-codec/commons-codec/1.10/commons-codec-1.10.jar
[info] Found library net.java.jinput:jinput:2.0.5 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/net/java/jinput/jinput/2.0.5/jinput-2.0.5.jar
[info] Found library net.java.jutils:jutils:1.0.0 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/net/java/jutils/jutils/1.0.0/jutils-1.0.0.jar
[info] Found library com.mojang:brigadier:0.1.27 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/mojang/brigadier/0.1.27/brigadier-0.1.27.jar
[info] Found library com.mojang:datafixerupper:1.0.3 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/mojang/datafixerupper/1.0.3/datafixerupper-1.0.3.jar
[info] Found library com.google.code.gson:gson:2.8.0 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/google/code/gson/gson/2.8.0/gson-2.8.0.jar
[info] Found library com.mojang:authlib:1.5.25 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/mojang/authlib/1.5.25/authlib-1.5.25.jar
[info] Found library org.apache.commons:commons-compress:1.8.1 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/apache/commons/commons-compress/1.8.1/commons-compress-1.8.1.jar
[info] Found library org.apache.httpcomponents:httpclient:4.3.3 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/apache/httpcomponents/httpclient/4.3.3/httpclient-4.3.3.jar
[info] Found library commons-logging:commons-logging:1.1.3 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar
[info] Found library org.apache.httpcomponents:httpcore:4.3.2 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar
[info] Found library it.unimi.dsi:fastutil:7.1.0 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/it/unimi/dsi/fastutil/7.1.0/fastutil-7.1.0.jar
[info] Found library org.apache.logging.log4j:log4j-api:2.8.1 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/apache/logging/log4j/log4j-api/2.8.1/log4j-api-2.8.1.jar
[info] Found library org.apache.logging.log4j:log4j-core:2.8.1 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/apache/logging/log4j/log4j-core/2.8.1/log4j-core-2.8.1.jar
[info] Found library org.lwjgl:lwjgl:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl/3.1.6/lwjgl-3.1.6-natives-macos.jar
[info] Found library org.lwjgl:lwjgl-jemalloc:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-jemalloc/3.1.6/lwjgl-jemalloc-3.1.6-natives-macos.jar
[info] Found library org.lwjgl:lwjgl-openal:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-openal/3.1.6/lwjgl-openal-3.1.6-natives-macos.jar
[info] Found library org.lwjgl:lwjgl-opengl:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-opengl/3.1.6/lwjgl-opengl-3.1.6.jar
[info] Found library org.lwjgl:lwjgl-glfw:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-glfw/3.1.6/lwjgl-glfw-3.1.6.jar
[info] Found library org.lwjgl:lwjgl-stb:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-stb/3.1.6/lwjgl-stb-3.1.6.jar
[info] Found library org.lwjgl:lwjgl:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl/3.1.6/lwjgl-3.1.6-natives-macos.jar
[info] Found library org.lwjgl:lwjgl-jemalloc:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-jemalloc/3.1.6/lwjgl-jemalloc-3.1.6-natives-macos.jar
[info] Found library org.lwjgl:lwjgl-openal:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-openal/3.1.6/lwjgl-openal-3.1.6-natives-macos.jar
[info] Found library org.lwjgl:lwjgl-opengl:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-opengl/3.1.6/lwjgl-opengl-3.1.6.jar
[info] Found library org.lwjgl:lwjgl-glfw:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-glfw/3.1.6/lwjgl-glfw-3.1.6.jar
[info] Found library org.lwjgl:lwjgl-stb:3.1.6 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/org/lwjgl/lwjgl-stb/3.1.6/lwjgl-stb-3.1.6.jar
[info] Found library com.mojang:text2speech:1.10.3 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/mojang/text2speech/1.10.3/text2speech-1.10.3.jar
[info] Found library com.mojang:text2speech:1.10.3 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/com/mojang/text2speech/1.10.3/text2speech-1.10.3.jar
[info] Found library ca.weblite:java-objc-bridge:1.0.0 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/ca/weblite/java-objc-bridge/1.0.0/java-objc-bridge-1.0.0-natives-osx.jar
[info] Found library ca.weblite:java-objc-bridge:1.0.0 at file:/Users/dmaxwell-ross/Library/Application%20Support/minecraft/libraries/ca/weblite/java-objc-bridge/1.0.0/java-objc-bridge-1.0.0-natives-osx.jar
[info] Finished loading libraries.
[info] Recognised Minecraft Version 1.13-pre5 with the magic string "adbxltvoh[Lsx;vj[J[[Jvfurupbvauqwvpmpqcbqpjpk".
[info] Reading 1.13-pre5.jar
[info] Jar load complete.
[info] Searching for classes...
[info] Found class Bootstrap: pa
[info] Found class LayerUtil: byx
[info] Found class GenLayer: byw
[info] Found class WorldType: axj
[info] Found class OverworldGenSettings: bns
[info] Found class Biome: axx
[info] Class search complete.
[info] Loading classes...
[info] Classes loaded.
[info] Minecraft load complete.
[info] Not writing to seed history file, because it does not exist: history.txt
[info] Creating world with seed '6015600670659945828' and type 'Default'
[info] Using the following generator options: 
[warning] amidst.mojangapi.minecraftinterface.MinecraftInterfaceException: unable to create world
	at amidst.mojangapi.minecraftinterface.local.LocalMinecraftInterface.createWorld(LocalMinecraftInterface.java:155)
	at amidst.mojangapi.minecraftinterface.LoggingMinecraftInterface.createWorld(LoggingMinecraftInterface.java:26)
	at amidst.mojangapi.world.WorldBuilder.create(WorldBuilder.java:114)
	at amidst.mojangapi.world.WorldBuilder.from(WorldBuilder.java:64)
	at amidst.mojangapi.RunningLauncherProfile.createWorld(RunningLauncherProfile.java:69)
	at amidst.gui.main.WorldSwitcher.displayWorld(WorldSwitcher.java:67)
	at amidst.gui.main.Actions.newFromSeed(Actions.java:77)
	at amidst.gui.main.Actions.newFromRandom(Actions.java:70)
	at amidst.gui.main.menu.Menus.lambda$item$2(Menus.java:133)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(Unknown Source)
	at java.desktop/javax.swing.AbstractButton.doClick(Unknown Source)
	at java.desktop/javax.swing.AbstractButton.doClick(Unknown Source)
	at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Actions.actionPerformed(Unknown Source)
	at java.desktop/javax.swing.SwingUtilities.notifyAction(Unknown Source)
	at java.desktop/javax.swing.JComponent.processKeyBinding(Unknown Source)
	at java.desktop/javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source)
	at java.desktop/javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source)
	at java.desktop/javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source)
	at java.desktop/javax.swing.JMenuBar.processKeyBinding(Unknown Source)
	at java.desktop/javax.swing.KeyboardManager.fireBinding(Unknown Source)
	at java.desktop/javax.swing.KeyboardManager.fireKeyboardAction(Unknown Source)
	at java.desktop/javax.swing.JComponent.processKeyBindingsForAllComponents(Unknown Source)
	at java.desktop/javax.swing.SwingUtilities.processKeyBindings(Unknown Source)
	at java.desktop/javax.swing.UIManager$2.postProcessKeyEvent(Unknown Source)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(Unknown Source)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(Unknown Source)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(Unknown Source)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Window.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue.access$600(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at amidst.clazz.symbolic.SymbolicMethod.invoke(SymbolicMethod.java:70)
	at amidst.clazz.symbolic.SymbolicMethod.callFromObject(SymbolicMethod.java:59)
	at amidst.clazz.symbolic.SymbolicMethod.callStatic(SymbolicMethod.java:52)
	at amidst.clazz.symbolic.SymbolicClass.callStaticMethod(SymbolicClass.java:93)
	at amidst.mojangapi.minecraftinterface.local.LocalMinecraftInterface.createWorld(LocalMinecraftInterface.java:140)
	... 52 more
Caused by: java.lang.ExceptionInInitializerError
	at pa.c(SourceFile:466)
	... 61 more
Caused by: java.util.ConcurrentModificationException
	at java.base/java.util.HashMap.computeIfAbsent(Unknown Source)
	at com.mojang.datafixers.DSL.taggedChoiceType(DSL.java:251)
	at com.mojang.datafixers.types.templates.TaggedChoice.lambda$null$1(TaggedChoice.java:72)
	at java.base/java.util.HashMap.computeIfAbsent(Unknown Source)
	at com.mojang.datafixers.types.templates.TaggedChoice.lambda$apply$2(TaggedChoice.java:71)
	at com.mojang.datafixers.types.templates.Named.lambda$apply$0(Named.java:43)
	at com.mojang.datafixers.types.templates.Check$1.apply(Check.java:45)
	at com.mojang.datafixers.types.templates.Sum$1.apply(Sum.java:50)
	at com.mojang.datafixers.types.templates.Sum$1.apply(Sum.java:50)
	at com.mojang.datafixers.types.templates.Sum$1.apply(Sum.java:50)
	at com.mojang.datafixers.types.templates.Sum$1.apply(Sum.java:50)
	at com.mojang.datafixers.types.families.RecursiveTypeFamily.lambda$null$1(RecursiveTypeFamily.java:107)
	at com.mojang.datafixers.types.templates.RecursivePoint$RecursivePointType.unfold(RecursivePoint.java:132)
	at com.mojang.datafixers.types.templates.RecursivePoint$RecursivePointType.findCheckedType(RecursivePoint.java:183)
	at com.mojang.datafixers.schemas.Schema.getType(Schema.java:80)
	at com.mojang.datafixers.schemas.Schema.findChoiceType(Schema.java:119)
	at com.mojang.datafixers.schemas.Schema.getChoiceType(Schema.java:111)
	at aav.getChoiceType(SourceFile:23)
	at com.mojang.datafixers.schemas.Schema.resolveEntityType(Schema.java:103)
	at aec$a.a(SourceFile:438)
	at aec.a(SourceFile:130)
	at aec.<clinit>(SourceFile:136)
	... 62 more

[info] Not writing to seed history file, because it does not exist: history.txt
[info] Creating world with seed '4745463005009064154' and type 'Default'
[info] Using the following generator options: 
[crash] Amidst has encounted an uncaught exception on the thread Thread[AWT-EventQueue-0,6,main]
[crash] java.lang.ExceptionInInitializerError
	at byx.<clinit>(SourceFile:25)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at amidst.clazz.symbolic.SymbolicMethod.invoke(SymbolicMethod.java:70)
	at amidst.clazz.symbolic.SymbolicMethod.callFromObject(SymbolicMethod.java:59)
	at amidst.clazz.symbolic.SymbolicMethod.callStatic(SymbolicMethod.java:52)
	at amidst.clazz.symbolic.SymbolicClass.callStaticMethod(SymbolicClass.java:93)
	at amidst.mojangapi.minecraftinterface.local.LocalMinecraftInterface.createWorld(LocalMinecraftInterface.java:143)
	at amidst.mojangapi.minecraftinterface.LoggingMinecraftInterface.createWorld(LoggingMinecraftInterface.java:26)
	at amidst.mojangapi.world.WorldBuilder.create(WorldBuilder.java:114)
	at amidst.mojangapi.world.WorldBuilder.from(WorldBuilder.java:64)
	at amidst.mojangapi.RunningLauncherProfile.createWorld(RunningLauncherProfile.java:69)
	at amidst.gui.main.WorldSwitcher.displayWorld(WorldSwitcher.java:67)
	at amidst.gui.main.Actions.newFromSeed(Actions.java:77)
	at amidst.gui.main.Actions.newFromRandom(Actions.java:70)
	at amidst.gui.main.menu.Menus.lambda$item$2(Menus.java:133)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(Unknown Source)
	at java.desktop/javax.swing.AbstractButton.doClick(Unknown Source)
	at java.desktop/javax.swing.AbstractButton.doClick(Unknown Source)
	at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Actions.actionPerformed(Unknown Source)
	at java.desktop/javax.swing.SwingUtilities.notifyAction(Unknown Source)
	at java.desktop/javax.swing.JComponent.processKeyBinding(Unknown Source)
	at java.desktop/javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source)
	at java.desktop/javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source)
	at java.desktop/javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source)
	at java.desktop/javax.swing.JMenuBar.processKeyBinding(Unknown Source)
	at java.desktop/javax.swing.KeyboardManager.fireBinding(Unknown Source)
	at java.desktop/javax.swing.KeyboardManager.fireKeyboardAction(Unknown Source)
	at java.desktop/javax.swing.JComponent.processKeyBindingsForAllComponents(Unknown Source)
	at java.desktop/javax.swing.SwingUtilities.processKeyBindings(Unknown Source)
	at java.desktop/javax.swing.UIManager$2.postProcessKeyEvent(Unknown Source)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(Unknown Source)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(Unknown Source)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(Unknown Source)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Window.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue.access$600(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: java.lang.IllegalStateException: Invalid Biome requested: ocean
	at ayc.a(SourceFile:92)
	at ayc.<clinit>(SourceFile:13)
	... 62 more
```

**Comment by moulins** [comment]
This is issue #395.

In 1.13 snapshots, Mojang completely reworked the biome generation code, and in doing so broke Amidst. If you know how to build Amidst from sources, the branch [1-13-snapshots](https://github.com/toolbox4minecraft/amidst/tree/1-13-snapshots) has a working implementation for the latest snapshots, but otherwise you'll have to wait for the next release. :)

**Comment by dmr-loggly** [comment]
@moulins, this was recorded on a build made from 1-13-snapshots; that's why I reported it. I knew the main line of development wasn't expected to work.

**Comment by moulins** [comment]
Oh, sorry.

I just tested it on my machine with the latest commit and everything works correctly. Which commit did you use specifically ?

**Comment by dmr-loggly** [comment]
I've just reproduced with `fd8c3d7a25c4d267efd143aaa1daaffe2f76fa11`

**Comment by dmr-loggly** [comment]
BTW, I have a working debug environment if you'd like me to collect any other information.

**Comment by moulins** [comment]
Definitely can't reproduce on my end. Maybe it is a Mac-only bug?

I'll have to try it on a Mac when I can.

**Comment by dmr-loggly** [comment]
OSX-only is definitely plausible... I do notice that both stack traces contain `doPrivileged(Native Method)` pretty close to the root.

I'll try things out on my Windows box too, when I get a chance.

**Comment by dmr-loggly** [comment]
I've just reproduced the issue on a Windows machine as well. Commit `fd8c3d7a25c4d2`

Perhaps this is an version issue of some sort? Here's what I'm using (on both machines):
```
 ~\amidst [1-13-snapshots ≡]> mvn --version
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T11:33:14-07:00)
Maven home: C:\Users\Daniel\apache-maven-3.5.4\bin\..
Java version: 10.0.1, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-10.0.1
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
~\amidst [1-13-snapshots ≡]> java --version
java 10.0.1 2018-04-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.1+10)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.1+10, mixed mode)
~\amidst [1-13-snapshots ≡]> javac --version
javac 10.0.1
~\amidst [1-13-snapshots ≡]>
```

**Comment by moulins** [comment]
Hmm, maybe the Minecraft JAR doesn't like using Java 10?

What happens when launching Amidst with the JRE embedded in Minecraft? (and what happens when launching Minecraft itself with Java 10?)

**Comment by dmr-loggly** [comment]
I don't know how to accomplish either of those off the top of my head. I can research it later, but if you have a how-to link it'd be appreciated

**Comment by dmr-loggly** [comment]
Setting "java executable" in the launcher profile to `/usr/bin/java` results in an immediate crash.

**Comment by moulins** [comment]
Also confirmed on my end with Java 9, on a Ubuntu machine; this is because Minecraft only supports Java 8.

If you try to launch Minecraft with Java 9, you get the following error (after removing the `-XX:+CMSIncrementalMode` option, which is no longer supported):
```
java.lang.ExceptionInInitializerError
	at pa.c(SourceFile:466)
	... 61 more
Caused by: java.util.ConcurrentModificationException
	at java.base/java.util.HashMap.computeIfAbsent(Unknown Source)
	at com.mojang.datafixers.DSL.taggedChoiceType(DSL.java:251)
	at com.mojang.datafixers.types.templates.TaggedChoice.lambda$null$1(TaggedChoice.java:72)
	at java.base/java.util.HashMap.computeIfAbsent(Unknown Source)
	at com.mojang.datafixers.types.templates.TaggedChoice.lambda$apply$2(TaggedChoice.java:71)
	at com.mojang.datafixers.types.templates.Named.lambda$apply$0(Named.java:43)
	at com.mojang.datafixers.types.templates.Check$1.apply(Check.java:45)
	at com.mojang.datafixers.types.templates.Sum$1.apply(Sum.java:50)
	at com.mojang.datafixers.types.templates.Sum$1.apply(Sum.java:50)
	at com.mojang.datafixers.types.templates.Sum$1.apply(Sum.java:50)
	at com.mojang.datafixers.types.templates.Sum$1.apply(Sum.java:50)
	at com.mojang.datafixers.types.families.RecursiveTypeFamily.lambda$null$1(RecursiveTypeFamily.java:107)
	at com.mojang.datafixers.types.templates.RecursivePoint$RecursivePointType.unfold(RecursivePoint.java:132)
	at com.mojang.datafixers.types.templates.RecursivePoint$RecursivePointType.findCheckedType(RecursivePoint.java:183)
	at com.mojang.datafixers.schemas.Schema.getType(Schema.java:80)
	at com.mojang.datafixers.schemas.Schema.findChoiceType(Schema.java:119)
	at com.mojang.datafixers.schemas.Schema.getChoiceType(Schema.java:111)
	at aav.getChoiceType(SourceFile:23)
	at com.mojang.datafixers.schemas.Schema.resolveEntityType(Schema.java:103)
	at aec$a.a(SourceFile:438)
	at aec.a(SourceFile:130)
	at aec.<clinit>(SourceFile:136)
	... 62 more
```

I'll close this issue, since this isn't something we can fix ourselves.

**Comment by moulins** [comment]
**Note:** I opened [this issue](https://bugs.mojang.com/browse/MC-132986) on the Minecraft bug tracker; let's hope that Mojang fixes it. :smiley:

**Comment by dmr-loggly** [comment]
Good news-- they resolved it @ 1.13-pre7
